src/ocamlutil/pretty.ml

```diff
@@ -70,30 +70,32 @@ type doc =
   | Unmark
 
 (* Break a string at \n *)
-let rec breakString (acc: doc) (str: string) : doc =
-  (* Printf.printf "breaking string %s\n" str; *)
-  match (try Some (String.index str '\n') with Not_found -> None) with
-  | None -> if acc = Nil then Text str else CText (acc, str)
-  | Some r ->
-    (* Printf.printf "r=%d\n" r; *)
-    let len = String.length str in
-    if r > 0 then begin
-      (* Printf.printf "Taking %s\n" (String.sub str 0 r); *)
-      let acc' = Concat(CText (acc, String.sub str 0 r), Line) in
-      if r = len - 1 then (* The last one *)
-        acc'
```

Comment:
The old logic avoided `Text ""` when the string ends with a newline as well.

Reply:
I have restored this behavior now.